### ISSUE TITLE
Custom Mono export crashes when using vanilla export templates

### ISSUE THREAD

**Issue by SimplyMiha** [godotengine/godot#41700]
**Godot version:**
Custom in-house build based on 3.2.3 beta 1

**OS/device including version:**
Windows 10

**Issue description:**
We are using an in-house Godot fork with some modifications necessary for our needs. Last week, we tested a HTML5 export, which prompted me to download export templates. Naturally, these turned out not to be compatible with our Godot version, so we dropped the HTML5 export until we would prepare a proper HTML5 export template.

However, since then, our regular Windows 10 builds started crashing on startup. After an investigation, I pinpointed the cause to installed export templates that apparently were still used by the build process despite using custom export templates for Windows 10 (custom template paths defined in export options).

Removing the export templates from appdata folder solved the issue; the custom templates are now used in the build process and the executable no longer crashes.

I opened this issue in order to document it.

**Steps to reproduce:**
- using a modified Godot version, export the project using custom templates; the executable runs without issues
- install an export templates package matching the base Godot version
- export the project using the same configuration; the executable crashes on startup, in my case with this error:
`Runtime critical type Mono.RuntimeStructs/HandleStackMark not found (in domain , error)`
- delete the installed export packages
- exported executable no longer crashes

**Minimal reproduction project:**
Not applicable

**Comment by akien-mga** [comment]
What's the full version of your custom editor build (`godot --version`)?
If it's not `3.2.3.beta1`, then it would not use the installed 3.2.3.beta1 export templates, the editor is very strict about this.

So the most likely is that the custom templates defined in your export preset were invalid, or you use another export preset that does not have those custom templates. From my knowledge of the export code, I don't see how the situation you describe could happen.

**Comment by SimplyMiha** [comment]
The full version is v3.2.3.beta.mono.custom_build.87b4bb801
It may have been the wrong export templates after all. Even then, custom templates were defined in export options for Windows 10 export - shouldn't the build process prioritize those?

**Comment by akien-mga** [comment]
> The full version is v3.2.3.beta.mono.custom_build.87b4bb801

Then it could only have used:

1. Custom export templates if they're defined.
2. Otherwise, templates in `%APPDATA%/Godot/templates/3.2.3.beta.mono/` (i.e. not `3.2.3.beta1.mono` templates). There hasn't been any official release named `3.2.3.beta.mono` so if you had this folder, you must have created it yourself.

Note that this logic is the same for Release and Debug builds, so if you had e.g. specified a custom templates only for Debug, but exported a Release build, it would pick the template in `%APPDATA%` if there's one for `3.2.3.beta.mono`.

**Comment by SimplyMiha** [comment]
Makes sense. Because there were no official export templates, I had taken the closest version (beta1) and renamed to satisfy the strict editor requirements. Since that didn't work, I abandoned the idea of HTML5 export for another time. Naturally, since I didn't remove the installed templates, the build process was still using them.

Custom templates are defined as .exe files of the custom build and were ignored for some reason when building (or so I am guessing, since builds stopped failing after I removed the export templates).

I'll be more cautious with export templates in the future. Thank you for your time!
